src/app/products/products.component.ts

```diff
@@ -1,7 +1,25 @@
+import { CategoryService } from './../category.service';
+import { ProductService } from './../product.service';
 import { Component } from '@angular/core';
 
 @Component({
   selector: 'app-products',
   templateUrl: './products.component.html'
 })
-export class ProductsComponent {}
+export class ProductsComponent {
+  products$;
```

Comment:
these should have types. also i could be wrong but aren't the variables with $ signs supposed to be of type observable?

Reply:
then you run em through angular's async pipe. 

`products$ = categoryService
      .getCategories()
      .valueChanges()` and in the html you would do` {product$ | async }`

Reply:
I will set up types tomorrow, but may need assistance with async method as the error says cannot run async on [object object] in console.  I will arrive tomorrow early and try to tackle again.